client/rest/src/plugins/rosetta/accountRoutes.js

```diff
@@ -52,43 +81,61 @@ export default {
 
 			const startChainHeight = await getChainHeight();
 
-			const mosaics = await services.proxy.fetch(`accounts/${address}`, json => json.account.mosaics);
+			const promises = [services.proxy.fetch(`accounts/${address}`, json => json.account.mosaics)];
+			if (typedRequest.include_mempool)
+				promises.push(services.proxy.fetchAll('transactions/unconfirmed?embedded=true', PAGE_SIZE));
+
+			const [mosaics, unconfirmedTransactions] = await Promise.all(promises);
 
 			const endChainHeight = await getChainHeight();
 
 			if (startChainHeight !== endChainHeight)
 				throw RosettaErrorFactory.SYNC_DURING_OPERATION;
 
+			let amounts = await mapMosaicsToRosettaAmounts(mosaics);
+			if (typedRequest.include_mempool && unconfirmedTransactions?.length) {
+				const combineAmountsValue = (lhs, rhs) => (BigInt(lhs) + BigInt(rhs)).toString();
+
+				// combine the operations by each currency
+				const currencyAmountMap = new Map();
+				const userOperations = await getAccountOperations(unconfirmedTransactions, address);
+				userOperations.forEach(userOperation => {
+					const storedValue = currencyAmountMap.get(userOperation.amount.currency.symbol)?.value || 0;
+					userOperation.amount.value = combineAmountsValue(userOperation.amount.value, storedValue);
```

Comment:
did you intend to update the userOperation value too? original version was only updating stored value